### ISSUE TITLE
Make the use of `BuildKit` possible in the image building step.

### ISSUE THREAD

**Issue by FeryET** [PrefectHQ/prefect-docker#56]
Hi.

Docker buildkit helps me to  build images much faster on my local PC, wanted to see if there is anyway to use the buildkit in the image building phase in prefect_docker?

I can help with the code if needed.

**Comment by ngriffiths13** [comment]
I second this. It would be great if there was a way to use buildkit.

**Comment by desertaxle** [comment]
Based on https://github.com/docker/docker-py/issues/2230 it looks like `docker-py` (the Docker library that `prefect-docker` uses) does not support BuildKit.  As a result, supporting BuildKit in `prefect-docker` will be difficult. One possible solution is to use another library like [pyton-on-whales](https://github.com/gabrieldemarmiesse/python-on-whales) that supports BuildKit, but migrating will require rewriting large portions of `prefect-docker`.

**Comment by desertaxle** [comment]
It looks like it is possible to set the `buildkit` flag at the daemon level: https://github.com/PrefectHQ/prefect/issues/10331#issuecomment-1662282377.

**Comment by FeryET** [comment]
I don't think this necessarily address the issue espeically in the code. Since `DOCKER_BUILDKIT=1` uses multithreaded build system and caching which AFAIK `docker-py` does not support. So regardless of the flag, I think it will not work.

I think supporting both `Python On Whales` and docker-py can be a good option. Since `python-on-whales` also supports podman containers, and most users do not want fine-grained control over what docker-daemon is doing like what Docker-py offers.
